tools/ci_build/github/linux/docker/scripts/training/azure_scale_set_vm_mount_test_data.sh

```diff
@@ -1,30 +1,70 @@
 #! /usr/bin/env bash
-if [ -d "/bert_data" ]; then
-sudo umount /bert_data
-fi
-if [ -d "/bert_data" ]; then
-sudo rm -rf /bert_data
-fi
 
-sudo mkdir /bert_data
+function credentialize () {
+    if [ ! -d "/etc/smbcredentials" ]; then
+    sudo mkdir /etc/smbcredentials
+    fi
 
-if [ ! -d "/etc/smbcredentials" ]; then
-sudo mkdir /etc/smbcredentials
-fi
+    if [ -f "/etc/smbcredentials/orttrainingtestdata.cred" ]; then
+    sudo rm /etc/smbcredentials/orttrainingtestdata.cred
+    fi
 
-if [ -f "/etc/smbcredentials/orttrainingtestdata.cred" ]; then
-sudo rm /etc/smbcredentials/orttrainingtestdata.cred
-fi
+    # to create orttrainingtestdata.cred, I have to do: 'sudo bash -c ...'
+    sudo bash -c 'echo "username=orttrainingtestdata" >> /etc/smbcredentials/orttrainingtestdata.cred'
 
-# to create orttrainingtestdata.cred, I have to do: 'sudo bash -c ...'
-sudo bash -c 'echo "username=orttrainingtestdata" >> /etc/smbcredentials/orttrainingtestdata.cred'
+    # $1 get removed (do defend injection attack?) if I do 'sudo bash -c...'
+    # to enable 'sudo echo...' I need to 'sudo chmod 777...' first.
+    sudo chmod 777 /etc/smbcredentials/orttrainingtestdata.cred
+    sudo echo "password=$1" >> /etc/smbcredentials/orttrainingtestdata.cred
 
-# $1 get removed (do defend injection attack?) if I do 'sudo bash -c...'
-# to enable 'sudo echo...' I need to 'sudo chmod 777...' first.
-sudo chmod 777 /etc/smbcredentials/orttrainingtestdata.cred
-sudo echo "password=$1" >> /etc/smbcredentials/orttrainingtestdata.cred
+    sudo chmod 600 /etc/smbcredentials/orttrainingtestdata.cred
+}
 
-sudo chmod 600 /etc/smbcredentials/orttrainingtestdata.cred
+function mount_mnist () {
+    echo "Mounting mnist"
 
-sudo bash -c 'echo "//orttrainingtestdata.file.core.windows.net/bert-data /bert_data cifs nofail,vers=3.0,credentials=/etc/smbcredentials/orttrainingtestdata.cred,dir_mode=0777,file_mode=0777,serverino" >> /etc/fstab'
-sudo mount -t cifs //orttrainingtestdata.file.core.windows.net/bert-data /bert_data -o vers=3.0,credentials=/etc/smbcredentials/orttrainingtestdata.cred,dir_mode=0777,file_mode=0777,serverino
+    if [ -d "/mnist" ]; then
+    sudo umount /mnist
+    fi
+    if [ -d "/mnist" ]; then
+    sudo rm -rf /mnist
```

Comment:
What is the goal here? Delete the folder if it fails to umount? Or delete if there is an existing folder at /mnist after unmounting?

If it is the latter, we need to check the exit code of umount to make a decision, otherwise we might delete the content of ountpoint by accident

Reply:
unmounting will unmount all the data in that directory. But will not delete that directory. So, if there was something mounted there previously, that directory will still exist after unmounting. In order to start from a clean state, this was done. This is what my understanding of this code is.

Reply:
But now, since I am taking in arguments to the destination directory, it may be unsafe to do this. Maybe I can mount the directory with non root permissions in a local directory (maybe the build directory). In that case, I won't accidentally remove important directories with ```sudo rm -rf```.

Reply:
Yes that is the case